docs/prompting/thought_generation/chain_of_thought_zero_shot/thread_of_thought.md

```diff
@@ -1,7 +1,128 @@
 ---
-title: ""
-description: ""
-keywords: ""
+description: "Thread of Thought helps models ignore irrelevant context in their prompt, improving overall response quality and relevance"
 ---
 
-[wip]
+By encouraging our model to examine each source in the provided context, we can help mitigate the impact of irrelevant context. This improves reasoning performance and the final output. This is known as Thread Of Thought <sup><a href="https://arxiv.org/pdf/2311.08734">1</a></sup>.
+
+We can implement Thread Of Thought using the following template.
+
+!!! example "Thread Of Thought template"
+
+    **[ Input Prompt ]**
+
+    Proceed through the context systematically, zeroing in on areas that could provide the answers we’re seeking
+
+We can implement this in Instructor as seen below
+
+```python hl_lines="53-54"
+import instructor
+from openai import OpenAI
+from pydantic import BaseModel, Field
+
+
+client = instructor.from_openai(OpenAI())
+
+
+class Response(BaseModel):
+    analysis: list[str] = Field(
+        ...,
+        description="""This is a list that should contain a single
+        explanation for a relevant source""",
+    )
+    answer: str = Field(
+        ..., description="""This is the answer to the user's question"""
+    )
+
+
+def get_response():
```

Comment:
name the function better, and make the sources and questions parameters so other people can actually use the code.